src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java

```diff
@@ -16,14 +16,21 @@
 interface ConditionMapper {
 
     /**
-     * Maps expressions in the given request for a virtual table to a request for the physical table,
-     * based on the virtual table's primary key.
+     * Maps the update expression, condition expression, and expression attributes in the given virtual update request
+     * to the corresponding physical update request.
      */
-    void apply(RequestWrapper request);
+    void applyForUpdate(RequestWrapper request);
```

Comment:
Handling update requests is the most difficult and has more complications: each assignment in the update expression can potentially fan out, and the update expression and filter expression may have overlapping field references. (Query requests also have possibly 2 expressions, but each can be translated independently, because the field references can't overlap -- I think.)

Reply:
By `filter expression` for updates do you mean `ConditionExpression`?

Reply:
Yes that's what I meant